src/Update/Nix/FetchGit/Utils.hs

```diff
@@ -83,3 +102,21 @@ parseISO8601DateToDay t =
   case parseTimeM False defaultTimeLocale "%Y-%m-%d" justDate of
     Nothing -> Left $ InvalidDateString t
     Just day -> pure day
+
+instance Show Warning where
```

Comment:
It's idiomatic to have GHC derive `Show`. The derived version has the nice property that it can be pasted into Haskell source to represent the shown value.

`show` as defined here could be renamed to `prettyShow`.